### PR TITLE
[ci][build] add tests for python only compilation

### PR DESCRIPTION
**Pull request by youkaichao**
add tests for https://github.com/vllm-project/vllm/pull/9621

cc @dtrifiro

### CODE REVIEW

File: .buildkite/test-pipeline.yaml [hunk 1]
```diff
@@ -50,9 +50,9 @@ steps:
   - tests/multimodal
   - tests/test_utils
   - tests/worker
-  - tests/test_lazy_torch_compile.py
+  - tests/standalone_tests/lazy_torch_compile.py
```
Comment (dtrifiro) [inline .buildkite/test-pipeline.yaml:53]: this seems unrelated?

Comment (youkaichao) [reply .buildkite/test-pipeline.yaml:53]: kind of, just moved it by the way.

**Review by dtrifiro — APPROVED** [review]
